### ISSUE TITLE
how to do Identity Mixing?

### ISSUE THREAD

**Issue by loboere** [ali-vilab/FlashFace#5]

**Comment by jshilong** [comment]
This code does not implement Identity Mixing. You can implement it by combining two ID results on https://github.com/ali-vilab/FlashFace/blob/a3afe3f5a6851104ae0895dfdb207cfbdc978ea8/flashface/all_finetune/models/reference_unet.py#L96 and https://github.com/ali-vilab/FlashFace/blob/a3afe3f5a6851104ae0895dfdb207cfbdc978ea8/flashface/all_finetune/models/reference_unet.py#L108

**Comment by loboere** [comment]
I would implement it but I don't understand what part of the code represents the id

**Comment by jshilong** [comment]
> I would implement it but I don't understand what part of the code represents the id

Thank you very much for your interest in our work. To understand the meaning of these tensors, you may need to set a breakpoint and refer to the article. The ID information is encoded in "cfx". If you want to mix the IDs, you should first obtain "ctx_id1" and "ctx_id2". Then, you can calculate "x" using the following formula:
```
x = x + (self.self_attn.self_attn_first(y, ctx_id1)) * self.share_cache['similarity'] + (1 - self.share_cache['similarity']) * self.self_attn.self_attn_first(y, ctx_id2)
```
if you encounter any difficulties, please don't hesitate to ask me for help.

**Comment by loboere** [comment]
I don't understand how to load two IDs, ctx was generated from y and y from x, I don't know what x is, could you give me some code to load two IDs?

**Comment by loboere** [comment]
Do you plan to publish the code for Identity Mixing?

**Comment by jshilong** [comment]
> Do you plan to publish the code for Identity Mixing?

Absolutely, this would be a modification to the existing code. I will add this function soon.

**Comment by loboere** [comment]
It's been a while, please add it as soon as possible

**Comment by loboere** [comment]
any news?

**Comment by zimenglan-sysu-512** [comment]
+1

**Comment by loboere** [comment]
please @jshilong  add it as soon as possible

**Comment by loboere** [comment]
any news @jshilong

**Comment by molo32** [comment]
@jshilong  please release  this feature
